### ISSUE TITLE
Workspaces are not moved and disapper when switching from single to multi monitor setup

### ISSUE THREAD

**Issue by zeltak** [i3/i3#2936]
Output of `i3 --moreversion 2>&- || i3 --version`:

> zeltak@zlap:~|⇒  i3 --moreversion 2>&- || i3 --version                                                          
> Binary i3 version:  4.14 (2017-09-04) © 2009 Michael Stapelberg and contributors                                
> Running i3 version: 4.14 (2017-09-04) (pid 1015)o abort…)                                                       
> Loaded i3 config: /home/zeltak/.i3/config (Last modified: Tue 12 Sep 2017 09:50:24 AM IDT, 10211 seconds ago)   
> The i3 binary you just called: /usr/bin/i3                                                                      
> The i3 binary you are running: i3  

- using Arch linux with up to date updates. 
- current i3 config: 
https://paste.xinu.at/9bqshU/

- current g.card info:
https://paste.xinu.at/kJC/

- link to similar looking issue:
https://github.com/i3/i3/issues/2201

URL to a logfile as per http://i3wm.org/docs/debugging.html:

 http://logs.i3wm.org/logs/5765486922956800.bz2

**What I did:**

when using i3 with multi monitors some work spaces just disappear when moving from the single initial login present monitor (laptop) to a multi screen setup.
in short: login to laptop with i3. all defined monitors are available. Then switch via a xrander script to a multi monitor setup:

````
#!/bin/sh
xrandr --output eDP-1 --off --output DP-2 --off --output DP-1 --off --output HDMI-1 --off --output DP-1-3 --off --output DP-1-2 --primary --mode 3440x1440 --pos 0x0 --rotate normal --output DP-1-1 --mode 2560x1440 --pos 3440x0 --rotate left & i3-msg restart
````
now ie workspace 2 is gone

**What I saw:**
workspace '2' disappeared , i cant use bind or `i3-msg workspace 2` to get to work space 2. i can see it present in rofi but its impossible to get to it through i3. 
also if i change an app to 'float' i can see the apps on work space '2' behind.

**What I expected instead:**
workspace 2 should not disappear

**Comment by Airblader** [comment]
Does this issue disappear when you use

    disable_randr15 yes

In your config?

**Comment by zeltak** [comment]
thx!

wow after a few harrowing days that solved it!

couldnt find any documentation on what that does though :)

thx alot again, would be happy to help test anything needed to solve this issue

best

Z

**Comment by Airblader** [comment]
That's because we don't intend this to be a feature per se. We switched to RandR 1.5 with i3 4.14 and implemented this switch exactly in case there are problems. We'll still have to fix this issue.

**Comment by belminf** [comment]
> We'll still have to fix this issue.

This is also happening to me:

```
$ i3 --moreversion 2>&- || i3 --version
Binary i3 version:  4.14-non-git © 2009 Michael Stapelberg and contributors
Running i3 version: 4.14-non-git (pid 582)rl-c to abort…)
Loaded i3 config: /home/belminf/.config/i3/config (Last modified: Sun 17 Sep 2017 10:04:23 PM EDT, 730 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

Adding `disable_randr15 yes` to config helped. Let me know if I could help with tracking down the problem.

**Comment by stapelberg** [comment]
In the provided logfile, searching for “RandR 1.5” yields only one result, suggesting that the log doesn’t contain any output changes.

Could either of you provide a logfile which contains just the bare minimum steps to reproduce this issue, ideally with a step-by-step description of what you did? Bonus points for reproducing the issue with the default config.

**Comment by usernamenumber** [comment]
I have been having this problem as well, only after upgrading from 4.11 (I'm on Ubuntu Xenial) to 4.14 via the i3 repo, and adding `disable_randr15 yes` to my config... I still have the problem. :(

Happy to provide any info that might help. This issue makes the difference between i3 being an absolute dream to work with and almost unusable for me, because I have to dock/undock my laptop frequently. 

Comments above mention a log file; how do I generate one? 

In the mean-time, heres the info I have:

```
$ i3 --moreversion
Binary i3 version:  4.14.1 (2017-09-24) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.14.1 (2017-09-24) (pid 30999)bort…)
Loaded i3 config: /home/local/ANT/smibrd/.i3/config (Last modified: Thu 09 Nov 2017 10:54:06 AM EST, 1057 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

xrandr output while connected to dock (laptop monitor eDP1 + two external monitors DP1-*):
```
$ xrandr
Screen 0: minimum 8 x 8, current 6560 x 1920, maximum 32767 x 32767
eDP1 connected 1920x1080+4640+0 (normal left inverted right x axis y axis) 344mm x 193mm
   1920x1080     60.05*+  59.93    40.04  
   1680x1050     59.95    59.88  
   1600x1024     60.17  
   1400x1050     59.98  
   1600x900      60.00  
   1280x1024     60.02  
   1440x900      59.89  
   1280x960      60.00  
   1368x768      60.00  
   1360x768      59.80    59.96  
   1152x864      60.00  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   640x360       60.00  
DP1 disconnected (normal left inverted right x axis y axis)
DP1-1 connected 1200x1920+0+0 left (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
   1920x1080     60.00  
   1600x1200     60.00  
   1680x1050     59.88  
   1280x1024     60.02  
   1280x960      60.00  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
   720x400       70.08  
DP1-2 connected primary 3440x1440+1200+0 (normal left inverted right x axis y axis) 800mm x 335mm
   3440x1440     59.97*+  49.99  
   2560x1440     59.95  
   2560x1080     60.00  
   1720x1440     60.00  
   1920x1080     60.00    50.00    59.94  
   1600x1200     60.00  
   1280x1024     75.02    60.02  
   1280x800      59.81  
   1152x864      75.00  
   1280x720      60.00    50.00    59.94  
   1024x768      75.03    60.00  
   800x600       75.00    60.32  
   720x576       50.00  
   720x480       60.00    59.94  
   640x480       75.00    60.00    59.94  
   720x400       70.08  
DP1-3 disconnected (normal left inverted right x axis y axis)
DP2 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
HDMI2 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
```

xrandr ourput while disconnected:
```
$ xrandr
Screen 0: minimum 8 x 8, current 6560 x 1920, maximum 32767 x 32767
eDP1 connected 1920x1080+4640+0 (normal left inverted right x axis y axis) 344mm x 193mm
   1920x1080     60.05*+  59.93    40.04  
   1680x1050     59.95    59.88  
   1600x1024     60.17  
   1400x1050     59.98  
   1600x900      60.00  
   1280x1024     60.02  
   1440x900      59.89  
   1280x960      60.00  
   1368x768      60.00  
   1360x768      59.80    59.96  
   1152x864      60.00  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   640x360       60.00  
DP1 disconnected (normal left inverted right x axis y axis)
DP1-1 disconnected (normal left inverted right x axis y axis)
DP1-2 disconnected primary (normal left inverted right x axis y axis)
DP1-3 disconnected (normal left inverted right x axis y axis)
DP2 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
HDMI2 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
  1920x1200 (0x129) 154.000MHz +HSync -VSync
        h: width  1920 start 1968 end 2000 total 2080 skew    0 clock  74.04KHz
        v: height 1200 start 1203 end 1209 total 1235           clock  59.95Hz
  3440x1440 (0x12c) 319.750MHz +HSync -VSync
        h: width  3440 start 3488 end 3520 total 3600 skew    0 clock  88.82KHz
        v: height 1440 start 1443 end 1453 total 1481           clock  59.97Hz
```

Command I run after disconnect:

```
$ xrandr --output eDP1 --auto --primary --output DP2 --off --output DP1-1 --off --output DP1-2 --off
```

I end up with the workspaces that were on my two other monitors inaccessible (or in some cases displayed across half of an accessible workspace, though I haven't seen that since the upgrade; can provide a screenshot of what I mean if it's any help though).

**Comment by stapelberg** [comment]
> Comments above mention a log file; how do I generate one?

Please see https://i3wm.org/docs/debugging.html
